doc/new-guidelines.md

```diff
@@ -0,0 +1,454 @@
+`re-frame.core/reg-event-db`.
+
+```clojure
+;; bad
+(re-frame/reg-event-fx
+ :wakuv2.ui/save-all-confirmed
+ (fn [{:keys [db] :as cofx}]
+   ...))
+
+;; good
+(fx/defn save-all
+  {:events [:wakuv2.ui/save-all-confirmed]}
+  [{:keys [db] :as cofx}]
+  ...)
+```
+
+### Registering top-level re-frame subscriptions
+
+Use `subs.root/reg-root-key-sub` to register top-level (root) subscriptions.
+Additionally, register root subscriptions in the `subs.root` namespace.
+
+```clojure
+;; bad
+(re-frame/reg-sub
+ :view-id
+ (fn [db]
+   (:view-id db)))
+
+;; good
+(reg-root-key-sub :view-id :view-id)
+```
+
+### Registering layer-3 subscriptions
+
+The majority of the subscriptions should be defined as [layer-3
+subscriptions](https://day8.github.io/re-frame/subscriptions/#the-four-layer)
+due to performance constraints.
+
+```clojure
+;; bad
+(re-frame/reg-sub
+ :ens/preferred-name
+ (fn [db]
+   (get-in db [:multiaccount :preferred-name])))
+
+;; good
+(re-frame/reg-sub
+ :ens/preferred-name
+ :<- [:multiaccount]
+ (fn [multiaccount]
+   (:preferred-name multiaccount)))
+```
+
+### Requiring quo2 components
+
+Never directly import a submodule of `quo2`, instead, use the provided
+`quo2.core` namespace.
+
+```clojure
+;; bad
+(ns my-namespace
+  (:require [quo2.components.icon :as icon]))
+
+(icon/icon :i/verified)
+
+;; good
+(ns my-namespace
+  (:require [quo2.core :as quo2]))
+
+(quo2/icon :i/verified)
+```
+
+### Require/import
+
+Prefer `:as` instead of `:refer`. There are exceptions to this rule, e.g. the
+test macros `deftest` and `is`, which are ubiquitous in the Clojure community.
+
+```clojure
+;; bad
+(ns status-im.utils.datetime
+  (:require [cljs-time.coerce :refer [from-long]]))
+
+;; good
+(ns status-im.utils.datetime
+  (:require [cljs-time.coerce :as time.coerce]))
+```
+
+### Accessibility labels
+
+Use keywords instead of strings. As a bonus, remember keywords are cached in
+memory.
+
+```clojure
+;; bad
+[text/text {:accessibility-label "profile-nickname"}
+ "Markov"]
+
+;; good
+[text/text {:accessibility-label :profile-nickname}
+ "Markov"]
+```
+
+### Icons
+
+Use the appropriate keyword qualification/namespace and don't directly require
+namespaces inside `quo2/components/`.
+
+```clojure
+;; bad
+(require '[quo2.components.icon :as icons])
+(icons/icon :main-icons2/verified)
+
+;; good
+(require '[quo2.core :as quo2])
+(quo2/icon :i/verified)
+```
+
+### Translations
+
+Prefer to use translation placeholders instead of creating multiple translation
+keywords and concatenating them into a single string.
+
+```clojure
+;; bad
+;; Assume the translation key is:
+;;   "biometric-auth-error": "Unable perform biometric authentication"
+(str (i18n/label :t/biometric-auth-error) "(" error-code ")")
+
+;; good
+;; Assume the translation key is:
+;;   "biometric-auth-error": "Unable perform biometric authentication ({{code}})"
+(i18n/label :t/biometric-auth-error {:code error-code})
+```
+
+## Project Structure
+
```

Comment:
Can we consider giving a quick high level overview of the project structure first.
e.g src/ quo2, common, react-native status_im etc and then show that tree first. it's much smaller and easier to understand. Then we can describe what lives in each place and show the expanded trees of each if neccesary.

Something along the lines of:

As the mobile application grows in size and complexity, new patterns emerge to
organize the codebase. The structure presented below is just an example of what
the codebase could look like, so not all pieces are presented here.

At the root level the project structure is aligned into modules and applications.

```
src
├── common
├── status_im
│   
├── quo2
│   
└── utils
```

Common - where core react native dependencies and other code used across modules and applications. (IMO utils, common and react-native should all be in here. They are both code that is common to quo2 and status_im)

Quo2 - the component library for Status IM mobile

Status_IM - The code for Status IM mobile application
Separated by contexts ()

Utils - a variety of code for both testing, configuration and other aspects (probably should be in common) 


An example view of the tree should will look like the following:
```
src

├── status_im
│  └── contexts
│      ├── browser/
│      ├── communities
│      │   ├── components/
│      │   ├── screens/
│      │   └── utils
│      │       └── example.cljs
│      ├── messaging/
│      ├── user_settings/
│      └── wallet/
│  ├── setup
│      ├── core.cljs
│      ├── events.cljs
│      ├── navigation.cljs
│      ├── reloader.cljs
│      └── test_runner.cljs
│  ├── subs
│      ├── ens.cljs
│      └── pairing.cljs
├── quo2
│   ├── components/
│   ├── foundations/
│   ├── screens/
│   └── theme.cljs
├── common
│  └── bottom_sheet.cljs
├── react_native
│   ├── gesture.cljs
│   └── platform.cljs
└── utils
    ├── number.cljs
    └── test_helpers.cljs
```

Reply:
I think it's a nice suggestion @J-Son89, I can give that a try. Curiously, the fact that your examples don't agree with the new project structure in the doc tells me that it might not be clear to everyone, so again, just reinforcing your feedback that this part could be clearer.

Reply:
yeah, it was my understanding that it should be like this. imo the structure described in the doc is not correct.

most importantly we should have the `status_im` folder, which is missing at the moment in the document version

Reply:
That's one good point, I don't think we need a status-im top level directory. In my perspective, everything that's not a vendored library is a status namespace. I understand the reasoning and the practices, but I only see it adding verbosity to namespaces for little practical value.

But having shared my opinion, I think I don't care much, if anyone feels strong about it I can totally change this right now for us.

Reply:
This PR is only concerned with documenting our conventions, not really establishing them. I'd suggest we move the discussion to the chat. I'm going to update the docs in a separate commit and share in our channel to get votes/opinions.

Reply:
"In my perspective, everything that's not a vendored library is a status namespace. I understand the reasoning and the practices, but I only see it adding verbosity to namespaces for little practical value."

My view on this is that we should aim to section off pieces of code to their functionality so that it is not being mixed together.

e.g if code in status_im application is being used within the quo2 library then that's an issue. The correct naming makes it easier for developers to understand what code should go where and confirm the agreements we as a team have made. Which it seems there is already some confusion on 😛 

I'm flexible either way but my opinion is that  in large projects it is better to be verbose.

Reply:
I can live with that choice to put most things inside `src/status_im/`. Thanks for sharing your reasoning. I'm updating the docs taking your opinion in consideration and if the team agrees awesome!